docs/source/l1_to_l2b_e2e.rst

```diff
@@ -0,0 +1,198 @@
+L1 to L2b end to end Test
+--------------------------
+
+This section demonstrates how to run an end-to-end test for processing Level 1 (L1) data into Level 2b (L2b) data using the ``corgidrp`` package.
+
+The test involves processing raw L1 data through calibration procedures and comparing the results against known TVAC L2b data. It also ensures calibration file creation and removal from the calibration database. Before running the test, ensure you have set up the necessary paths to your data and calibration files. The following example demonstrates the steps involved in processing the data and running the test.
+
+
+Running the Test
+~~~~~~~~~~~~~~~~
+
+There are several ways to run the L1 to L2b end-to-end test:
+
+1. Using pytest
+   
+   .. code-block:: bash
+
+      # From the root directory of corgidrp
+      pytest tests/test_l1_to_l2b.py -v
+
+2. Direct execution
+   
+   .. code-block:: bash
+
+      # Run the script directly with default paths
+      python tests/test_l1_to_l2b.py
+
+      # Or specify custom paths
+      python tests/test_l1_to_l2b.py --tvacdata_dir /path/to/CGI_TVAC_Data --outputdir /path/to/output
+
+
+Test Setup
+~~~~~~~~~~
+
+To run this test, you will need three specific datasets from the database:
+
+1. **L1 Data** - Raw Level 1 data files.
+2. **L2b Data** - Processed Level 2b data files.
+3. **Cals** - Calibration files required for processing.
+
+If you don't have all the datasets downloaded, make sure you at least have these three files to run the test promptly. The script imports necessary libraries and determines the directory containing the script.
+   
+.. code-block:: python
+
+    import argparse
+    import os
+    import pytest
+    import numpy as np
+    import astropy.time as time
+    import astropy.io.fits as fits
+    import corgidrp
+    import corgidrp.data as data
+    import corgidrp.mocks as mocks
+    import corgidrp.walker as walker
+    import corgidrp.caldb as caldb
+    import corgidrp.detector as detector
+
+    thisfile_dir = os.path.dirname(__file__)  # this file's folder
+
+The `test_l1_to_l2b` function initializes paths for input data, calibration files, and output directories.
+
+.. code-block:: python
+
+    @pytest.mark.e2e
+    def test_l1_to_l2b(tvacdata_path, e2eoutput_path):
+        # Define input data paths
+        l1_datadir = os.path.join(tvacdata_path, "TV-36_Coronagraphic_Data", "L1")
+        l2b_datadir = os.path.join(tvacdata_path, "TV-36_Coronagraphic_Data", "L2b")
+        processed_cal_path = os.path.join(tvacdata_path, "TV-36_Coronagraphic_Data", "Cals")
```

Comment:
The calibrations are complicated... The corgidrp will be producing the calibrations as well for the real data. We don't have that for test data here, so we are hacking the processed cals for the test data and squeezeing them into the official corgidrp cal format. So we should really caution that the code here is for the example TVAC data. A user of real data would either 

1) Have downloaded processed calibration FITS files, and just need to add them to the caldb.
2) Have processed the calibrations themselves on the same computer. In this case, the calibration file should already be in their caldb file (located in `~/.corgidrp/`)

Reply:
Did you address this comment? I'm not sure I can find additional clarification in the tutorial.